juju/juju
# Set target series in pre-upgrade-series hook environment

Adds `JUJU_TARGET_SERIES` to environment when running a `pre-upgrade-series` hook.

## QA steps

- Bootstrap this patch, ensuring debug log-level.
- `juju deploy ./acceptancetests/repository/charms/dummy-sink --series xenial`
- `juju upgrade-series 0 prepare bionic -y`
- Check the `debug-log` output and see that the charm's pre-upgrade-series hook prints its environment with the target series:
```
unit-dummy-sink-0: 12:09:21 DEBUG unit.dummy-sink/0.pre-series-upgrade environ({'DEBIAN_FRONTEND': 'noninteractive', 'JUJU_AGENT_SOCKET_ADDRESS': '@/var/lib/juju/agents/unit-dummy-sink-0/agent.socket', 'JUJU_AVAILABILITY_ZONE': '', 'APT_LISTCHANGES_FRONTEND': 'none', 'JUJU_CHARM_FTP_PROXY': '', 'JUJU_CHARM_HTTPS_PROXY': '', 'JUJU_PRINCIPAL_UNIT': '', 'JUJU_HOOK_NAME': 'pre-series-upgrade', 'JUJU_CONTEXT_ID': 'dummy-sink/0-pre-series-upgrade-6644375957170369267', 'JUJU_MACHINE_ID': '0', 'JUJU_MODEL_NAME': 'default', 'CHARM_DIR': '/var/lib/juju/agents/unit-dummy-sink-0/charm', 'JUJU_METER_STATUS': 'AMBER', 'JUJU_API_ADDRESSES': '10.161.87.106:17070', 'JUJU_CHARM_HTTP_PROXY': '', 'CLOUD_API_VERSION': '', 'TERM': 'tmux-256color', 'JUJU_TARGET_SERIES': 'bionic', 'JUJU_UNIT_NAME': 'dummy-sink/0', 'JUJU_SLA': 'unsupported', 'JUJU_VERSION': '2.9.9.1', 'JUJU_CHARM_NO_PROXY': '127.0.0.1,localhost,::1', 'JUJU_CHARM_DIR': '/var/lib/juju/agents/unit-dummy-sink-0/charm', 'LANG': 'C.UTF-8', 'JUJU_AGENT_SOCKET_NETWORK': 'unix', 'JUJU_METER_INFO': 'not set', 'JUJU_DISPATCH_PATH': 'hooks/pre-series-upgrade', 'PATH': '/var/lib/juju/tools/unit-dummy-sink-0:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin', 'JUJU_MODEL_UUID': 'c6f6dcfb-e51f-4e56-8f79-c5c594844d97'})
```

## Documentation changes

TBC

## Bug reference

https://bugs.launchpad.net/juju/+bug/1908035
